partner_country_lang/models/res_partner.py

```diff
@@ -0,0 +1,15 @@
+# Copyright 2022-2025 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class ResPartner(models.Model):
+    _inherit = "res.partner"
+
+    lang = fields.Selection(compute="_compute_lang")
+
+    @api.depends("country_id")
+    def _compute_lang(self):
+        for item in self:
+            item.lang = item.country_id.lang or item.lang
```

Comment:
Isn't more declarative this way and avoid unneeded assignations?

```suggestion
            if item.country_id.lang:
                item.lang = item.country_id.lang
```